internal/api/server.go

```diff
@@ -2,107 +2,220 @@ package api
+				},
+			}, nil
+		}
+		return nil, fmt.Errorf("failed to get config map: %w", err)
 	}
 
-	responseStruct := v1.ProbesArrayResponse{
-		Probes: []v1.ProbeObject{
-			dummyProbe,
-		},
+	probe := v1.ProbeObject{}
+	err = json.Unmarshal([]byte(cm.Data["probe-config.json"]), &probe)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal probe from configmap: %w", err)
 	}
 
-	return v1.GetProbeById200JSONResponse(responseStruct), nil
+	return v1.GetProbeById200JSONResponse(probe), nil
 }
 
 // (POST /metrics/probes)
-func (Server) CreateProbe(ctx context.Context, request v1.CreateProbeRequestObject) (v1.CreateProbeResponseObject, error) {
+func (s Server) CreateProbe(ctx context.Context, request v1.CreateProbeRequestObject) (v1.CreateProbeResponseObject, error) {
+	// Validate that a probe with this static_url doesn't already exist
+	urlHash := sha256.Sum256([]byte(request.Body.StaticUrl))
+	// A SHA256 hash is 64 characters, but Kubernetes labels are limited to 63.
+	// Truncating the hash is safe enough for this purpose.
+	urlHashString := hex.EncodeToString(urlHash[:])[:63]
+	hashLabelSelector := fmt.Sprintf("rhobs-synthetics/static-url-hash=%s", urlHashString)
+
+	existingProbes, err := s.KubeClient.CoreV1().ConfigMaps(s.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: hashLabelSelector,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to check for existing probes: %w", err)
+	}
+
+	if len(existingProbes.Items) > 0 {
+		return v1.CreateProbe409JSONResponse{
+			Error: &struct {
+				Code    *int32  `json:"code,omitempty"`
+				Message *string `json:"message,omitempty"`
+			}{
+				Code:    int32p(409),
+				Message: stringp(fmt.Sprintf("a probe for static_url %q already exists", request.Body.StaticUrl)),
+			},
+		}, nil
+	}
 
-	createdProbe := v1.ProbeObject{
-		Id:                  request.Body.ClusterId,
-		ApiserverUrl:        request.Body.ApiserverUrl,
-		ManagementClusterId: request.Body.ManagementClusterId,
-		Private:             request.Body.Private,
+	probeID := uuid.New()
+
+	// Prepare the probe object that will be stored
+	probeToStore := v1.ProbeObject{
+		Id:        probeID,
+		StaticUrl: request.Body.StaticUrl,
+		Labels:    request.Body.Labels,
 	}
 
-	responseBody := v1.ProbesArrayResponse{
-		Probes: []v1.ProbeObject{createdProbe},
+	// Convert payload to JSON for storing in the configmap
+	payloadBytes, err := json.Marshal(probeToStore)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal payload: %w", err)
 	}
 
-	log.Printf("Successfully created probe for cluster ID: %s", createdProbe.Id)
-	return v1.CreateProbe201JSONResponse(responseBody), nil
+	configMapName := fmt.Sprintf("probe-config-%s", probeID)
+
+	// Use labels from the request, and add our own app label
+	cmLabels := make(map[string]string)
+	if request.Body.Labels != nil {
+		for k, v := range *request.Body.Labels {
+			cmLabels[k] = v
+		}
+	}
+	cmLabels["app"] = "rhobs-synthetics"
```

Comment:
This appears to be the default label we rely on to gather probes during LIST operations too, correct?

If so, can we define this key+value as a constant and use that instead of hardcoding in each function?